typing/typedecl.ml

```diff
@@ -1097,10 +1111,12 @@ let transl_type_decl env rec_flag sdecl_list =
       Uid.mk ~current_unit:(Env.get_current_unit ())
     ) sdecl_list
   in
+  (* Additional scope since update_type may lower some generic levels *)
+  Ctype.with_local_level_generalize begin fun () ->
```

Comment:
@goldfirere and myself asked why this generalization region is not inside `update_type`, or maybe at least around the `List.iter2 (...)` below. Another option @goldfirere suggested was to define update_type to work only with generic types (newgenvar, newgenconstr).